hack/e2e-common.sh

```diff
@@ -34,7 +34,7 @@ function cluster_cleanup {
         $KIND export logs $ARTIFACTS --name $1 || true
         kubectl describe pods -n kueue-system > $ARTIFACTS/$1-kueue-system-pods.log || true
         kubectl describe pods > $ARTIFACTS/$1-default-pods.log || true
-        $KIND delete cluster --name $1
+        $KIND delete cluster -v=3 --name $1 || echo "Ignoring cluster delete failure."
```

Comment:
Can you point out another example where we ignore such a failure?

Reply:
the three lines above

Reply:
No strong view, but maybe it makes sense to only merge the `-v3` so that we can get more input if this repeats?
Otherwise we will not know if this happened.

Reply:
Unfortunately I doubt that  `-v=3` will make much of a difference (based on how I see the implementation in kind).

Reply:
So I see two paths forward:
1. only increase log verbosity now (the level TBD, maybe v5 is better?), and wait for repeating to see if we can understand it
2. ignore the errors as in the current version of the PR

I would be slightly leaning towards 1, as we are not under any pressure (the flake is rare), and it would be nice to understand what is going on.

OTOH I'm ok to accept as is, if this is @tenzen-y preference.

Reply:
Also, a related question, how long does the `kind delete cluster` normally take? 

If the time is normally within 10-20s, then it is likely that occasionally it just takes 30s. If this is normally 5s then I suspect there might be some resource leak, or a bug in kind we could report.

Reply:
The timeout is for killing the docker process, is not part of our setup, it's part of the image used to run the test in the CI, it's failure is already ignored.

Kind is just dropping any output of the `docker delete ` command and only prints the exit code.

In kind,  I was not able to find any `log.V(n)` where n > 3.

Reply:
I see, how long does it normally take to stop the docker container? 

Could it be that a bug in kind makes it exceptionally long sometimes? If so, maybe we can open a bug in kind and see if other folks are are of the problem.

Reply:
Docker fails to delete a container, kind only calls docker.

Why is deleting the cluster, in this context, a lot more relevant then exporting the logs?
(None of them are relvant to the kueue implementation)

Reply:
I would prefer to select opt 1 since once we omit errors, we never find errors in the phase of the deletion cluster.
I understand that the current deletion error is irrelevant to Kueue, but there is no guarantee for the future.

> only increase log verbosity now (the level TBD, maybe v5 is better?), and wait for repeating to see if we can understand it

My only concern for the increasing log level is huge log files.
IIRC, previously, we faced flaky tests due to the huge size of the log file.

Reply:
We are talking about `kind delete` v option which, based on what I can see in the kind implementation will not do too much.

And once again, there are no logs in kind done with a level higher than 3.

Reply:
> Why is deleting the cluster, in this context, a lot more relevant then exporting the logs?

If there is a bug to archive logs we only impact Kueue, with a bug to delete a cluster we may impact the infra.  I'm concerned with the risk of leaving clusters.

For example, if we silence the errors, and there is even a typo in the command to delete a cluster we would be  leaving orphan clusters, no? Maybe we could consider retry of the delete, but I'm not sure if the gain / complication ratio makes worth it.

I searched for similar issues and found https://github.com/helm/kind-action/issues/54, which leads to https://github.com/kubernetes-sigs/kind/issues/708. So, it seems this might be some issue with docker, where restart helps.

Reply:
> If there is a bug to archive logs we only impact Kueue,

How, it could be some kind or docker issue for all that we know. 

>  with a bug to delete a cluster we may impact the infra. I'm concerned with the risk of leaving clusters.

The test image also try to kill the docker daemon at this end of the test but if that fails it is just ignored.

Reply:
> The test image also try to kill the docker daemon at this end of the test but if that fails it is just ignored.

could you point out some code to support this?

Reply:
[https://github.com/kubernetes/test-infra/pull/31534, this is just adding the timeout, the ignore was there bef](https://github.com/dims/test-infra/blob/369faafcf2b2452c3afdac967a024cc4cdd941b2/images/bootstrap/runner.sh#L28-L35)

Reply:
This is actually dims's fork, but similar code is [here](https://github.com/kubernetes/test-infra/blob/master/images/bootstrap/runner.sh). 

@BenTheElder as the author of some of the code around there, would you say that it is safe to ignore the cluster failure delete in Kueue?

Reply:
it's not the end of the world to ignore but ...

Cluster delete failure should not be happening. Cluster deletion is reentrant and will succeed if there is nothing to do.

Reply:
The linked kind issue is really old, do we have current examples?

Reply:
>Cluster delete failure should not be happening. 

Yeah, but they do, as shown in the example. I'm not sure about the frequency, 

> Cluster deletion is reentrant and will succeed if there is nothing to do.

So, we may expect there is a good chance a retry would help? This might be safer, but it is a code complication, so I'm not convinced either.

Maybe we just park it and only proceed with ignoring if this repeats, so that we see the frequency? Did you see this before @trasc or @tenzen-y ?

Reply:
> Kind is just dropping any output of the docker delete  command and only prints the exit code.

that's not accurate. in kind exec errors contain the command output (which is always captured) and it is being printed in the error message ...

Reply:
> > Kind is just dropping any output of the docker delete  command and only prints the exit code.
> 
> that's not accurate. in kind exec errors contain the command output (which is always captured) and it is being printed in the error message ...

Indeed, I missed that.

> Cluster delete failure should not be happening. Cluster deletion is reentrant and will succeed if there is nothing to do.

It is, more or less, happening in our case as `cleanup_dind()` is able to delete all the containers.

```
Waiting 30 seconds for pods stopped with terminationGracePeriod:30
Cleaning up after docker
ed68da3fb667
6beb571d417e
bf442dfcffc1
...
```

> Did you see this before @trasc or @tenzen-y ?

No